var/spack/repos/builtin/packages/cpio/package.py

```diff
@@ -19,7 +19,8 @@ class Cpio(AutotoolsPackage, GNUMirrorPackage):
 
     version('2.13', sha256='e87470d9c984317f658567c03bfefb6b0c829ff17dbf6b0de48d71a4c8f3db88')
 
-    patch('https://src.fedoraproject.org/rpms/cpio/raw/dfe64c466d3ea2c8dfbd99700d9006f610064167/f/cpio-2.13-mutiple-definition.patch', sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b', when='%gcc@10:')
+    for ccomp in ('%gcc@10:', '%clang'):
+        patch('https://src.fedoraproject.org/rpms/cpio/raw/dfe64c466d3ea2c8dfbd99700d9006f610064167/f/cpio-2.13-mutiple-definition.patch', sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b', when=ccomp)
```

Comment:
That line is really long, and the CI failed due to raja marked as failing, so we've to re-run the the GitLab CI anyways.

What about reformatting it like this and then, rebasing it to the current develop branch which has the fix for raja:
```suggestion
        patch('https://src.fedoraproject.org/rpms/cpio/raw/dfe64c466d3ea2c8dfbd99700d9006f610064167/f/cpio-2.13-mutiple-definition.patch', sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b', when=ccomp)
               sha256='d22633c368b8aedf4c08b23b6fbaa81a52404c8943ab04926404083ac10f1a4b',
               when=ccomp)
```
A new commit (in that case rebased to the latest develop with raja fixed) fixed GitLab CI failing due to raja in this PR: #27167